paddle/cinn/common/integer_set.cc

```diff
@@ -78,6 +86,14 @@ std::optional<bool> SymbolicExprAnalyzer::ProveNE(const ir::Expr& lhs,
   if (diff.is_constant()) {
     return diff.get_constant() != 0;
   }
+  ir::Expr diff_lower_bound = LowerBound(diff);
```

Comment:
Can we use ProveEQ to  implement ProveNE?

ProveEQ returns true <==> ProveNE returns false
ProveEQ returns false <==> ProveNE returns true
ProveEQ returns optional null <==> ProveNE return optional null

Reply:
Modified.